count-stats/src/test/java/uk/ac/ebi/eva/countstats/integration/CountStatsIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+package uk.ac.ebi.eva.countstats.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import uk.ac.ebi.eva.countstats.model.Count;
+import uk.ac.ebi.eva.countstats.repository.CountRepository;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Testcontainers
+@ContextConfiguration(initializers = CountStatsIntegrationTest.DockerPostgreDataSourceInitializer.class)
+public class CountStatsIntegrationTest {
+    @Autowired
+    private MockMvc mvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private CountRepository countRepository;
+
+    public static PostgreSQLContainer<?> postgreDBContainer = new PostgreSQLContainer<>("postgres:9.6");
+
+    static {
+        postgreDBContainer.start();
+    }
+
+    public static class DockerPostgreDataSourceInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+        @Override
+        public void initialize(ConfigurableApplicationContext applicationContext) {
+            TestPropertySourceUtils.addInlinedPropertiesToEnvironment(
+                    applicationContext,
+                    "spring.datasource.url=" + postgreDBContainer.getJdbcUrl(),
+                    "spring.datasource.username=" + postgreDBContainer.getUsername(),
+                    "spring.datasource.password=" + postgreDBContainer.getPassword()
+            );
+        }
+    }
+
+    @Test
+    public void testSaveCount() throws Exception {
```

Comment:
Is there a way to clean up database contents after every tests so that if we add other tests later they don't interfere with each other?

Reply:
updated